packages/compartment-mapper/src/bundle.js

```diff
@@ -248,11 +255,155 @@ export const makeBundle = async (read, moduleLocation, options) => {
     module.bundlerKit = getBundlerKitForModule(module);
   }
 
+  return { modules, parsersInUse };
+};
+
+function wrapFunctorInPrecompiledModule (functorSrc, moduleIndex) {
+  const wrappedSrc = (
+`(function(){
+  with (this.scopeTerminator) {
+  with (this.globalThis) {
+    return function() {
+      'use strict';
+      return (
+${functorSrc}
+      );
+    };
+  }
+  }
+}).call(getEvalKitForModule(${moduleIndex}))()`)
+  return wrappedSrc
+}
```

Comment:
We could reuse the original wrappings by importing the evaluator making funciton and calling .toString() on it, then splitting on the line with eval.
I'm not sure if it's not too much work to be worth it, but it'd guarantee compatibility between the two methods as SES develops.

Reply:
i was thinking that if we wanted to keep the copy pasta for someone reason but still wanted to ensure it matches ses that we could have a test that checks the ses version and fails if it doesnt match expectations